frontend/src/components/Welcome.js

```diff
@@ -25,39 +26,210 @@ class Search extends Component {
       location: null,
+      viewport: null,
       loggedIn: false,
-      plusCode: '847X884Q+22'
+      openFind: true,
+      openCreate: false,
+      openShare: false,
+      plusCode: '847X884Q+22',
+      background: "",
+      playBottomAnim: "",
     };
   }
  
+  // Handles scrolling navigation so to handle background color changes
+  handleNavigation = (e) => {
+    const window = e.currentTarget;
+
+    // condition for if on the last section
+    let greaterThanNext = window.scrollY > document.body.scrollHeight - document.body.scrollHeight/2;
+    // condition for if on the second section
+    let lessThanNext = window.scrollY < document.body.scrollHeight - document.body.scrollHeight/2;
+    // condition for if on the first section
+    let greaterThanPrevious = window.scrollY > document.body.scrollHeight/4;
+
+    // Check scrolling direction
+    // This is going upwars
+    if (this.prev > window.scrollY) {
+      if (window.scrollY < document.body.scrollHeight/4) {
+	// Set the state to do an animation from dark to light
+	this.setState({background: "backgroundToLight"})
+      } else if (lessThanNext && greaterThanPrevious) {
+	// Set the state to do an animation from dark to dark
+	this.setState({background: "backgroundToDark"})
+      }
+    } else if (this.prev < window.scrollY) {
+      // Going downwards
+      if (lessThanNext && greaterThanPrevious) {
+	this.setState({background: "backgroundToDark"})
+      } else if (greaterThanNext) {
+	this.setState({background: "backgroundToLight"})
+      }
+    }
+    this.prev = window.scrollY;
+  };
   
-  render() {
-    return ( 
-      <div className="welcomeContent">
-        <Script url={url} onLoad={this.handleScriptLoad}/> 
-        <TopNavbar history={this.props.history}/>
-        <Jumbotron >
-          <h1> Welcome to MapIT! </h1> 
-          <Form>
-            <Form.Group>
-            <Form.Label> <h4>Our Mission</h4> </Form.Label>
-      	    <br/>
-            <Form.Label>The objective of MapIT is to manage and visualize student-created public events, and expose these events to their college community. <br />
-              We are here to help you explore campus events beyond your social circle.</Form.Label> <br/><br/>
-            <Form.Label> Enter your university </Form.Label> 
+  componentDidMount() {
+    this.prev = window.scrollY;
+    window.addEventListener('scroll', e => this.handleNavigation(e));
+  }
+
+  onFindClick() {
+    this.setState({
+      openFind: true,
+      openCreate: false,
+      openShare: false
+    });
+  }
+
+  onCreateClick() {
+    this.setState({
+      openFind: false,
+      openCreate: true,
+      openShare: false
+    });
+  }
+
+  onShareClick() {
+    this.setState({
+      openFind: false,
+      openCreate: false,
+      openShare: true
+    });
+  }
+
+  renderCollapses() {
+    return(
+      <div style={{width: "50%", textAlign: "left"}}>
+      	<h1 className="collapseTitle" onClick={this.onFindClick}>Find</h1>
```

Comment:
I know it's been tough to keep everything formatted nicely but the indentation in this file is super crazy!